apps/evm/src/hooks/useOperationModal/Modal/SupplyForm/index.tsx

```diff
@@ -569,9 +558,11 @@ const SupplyForm: React.FC<SupplyFormProps> = ({ asset, pool, onCloseModal }) =>
       }
 
       if (isWrappingNativeToken) {
-        return wrapTokensAndSupply({
+        return supply({
           accountAddress: accountAddress || '',
           amountMantissa,
+          wrap: true,
+          poolComptrollerContractAddress: pool.comptrollerAddress,
```

Comment:
Shouldn't this be unified with the call on line 557? Then we'd pass `wrap: isWrappingNativeToken`

Reply:
The input type of the `supply` function is quite restrictive (so that it can handle all flows), hence why it's not unified with line 567 as otherwise TypeScript will complain that certain properties passed don't exist on the corresponding input type.